server/plugin.go

```diff
@@ -289,7 +291,14 @@ func (p *Plugin) AddAutolinks(key, baseURL string) error {
 }
 
 func (p *Plugin) GetPluginKey() string {
-	return "mattermost_" + regexpNonAlnum.ReplaceAllString(p.GetSiteURL(), "_")
+	var k string
+	k = "mattermost_" + regexpNonAlnum.ReplaceAllString(p.GetSiteURL(), "_")
+	if len(k) <= 32 {
+		return k
+	}
+
+	k = uuid.New().String()
```

Comment:
I am not sure this would work as intended. As I remember, Plugin Key is supposed to be immutable, identifying the app in Atlassian. Can you please verify its uses?

Reply:
`Description: A unique key to identify the app.This key must be <= 64 characters.`
That's true. It should be unique. 

My workaround this is to use `KVSet` and then retrieve that every time. Thoughts?

Reply:
0/5 The intent here was to return a stable string, that would be reasonably unique and yet somewhat user-friendly if ever displayed. It may be sufficient to perhaps return `__` + the last 30 characters of the (escaped) site URL?

A KV store record seems an overkill.

Reply:
Hello @levb changed the implementation. The Jira docs say the key can be up to 64 characters long. Should we consider a bigger set if characters?